test/common/run-tests

```diff
@@ -67,83 +66,85 @@ def print_test(test, print_tap=True, retry_reason=""):
                 line = line[e.characters_written:]
                 time.sleep(0.1)
 
+    if retry_reason:
+        retry_reason = " " + retry_reason
+    if skip_reason:
+        skip_reason = " " + skip_reason
+
+
     if not print_tap:
-        print(retry_reason)
+        print((retry_reason or "") + (skip_reason or ""))
         flush_stdout()
         return
 
     print()  # Tap needs to start on a separate line
     if test.process.returncode == 0:
-        print("ok " + test_name(test) + retry_reason)
-    elif test.process.returncode == 77 or b"# SKIP " in test.output:
-        # If the test was skipped, add the last line (which contains the reason
-        # for the skip) to the result
-        print("ok {0} {1}{2}".format(test_name(test),
-                                     test.output.splitlines()[-1].strip().decode() if test.process.returncode == 77 else "",
-                                     retry_reason))
+        print("ok {0}{1}".format(test_name(test), retry_reason))
```

Comment:
Proposal for cleanup: do the ` # RETRY <reason>` formatting here instead of in teh caller. Then the whole line formatting is in one place, and you don't have to duplicate it any more. Same for skip, of course.

Reply:
The problem is with SKIP as we can get it from `test/common/testlib.py`. That means we have it once pre-formatted. Retry could be formatted here, but then it would be treated differently from skip.